src/components/EntryList.jsx

```diff
@@ -1,176 +1,93 @@
 import React, { useState, useEffect } from 'react';
-import { GeoFirestore } from 'geofirestore';
 import { useTranslation } from 'react-i18next';
 import { useHistory, useLocation } from 'react-router-dom';
-import * as Sentry from '@sentry/browser';
-import fb from '../firebase';
 import NotifyMe from './NotifyMe';
 import Entry from './entry/Entry';
 import Slider from './Slider';
 import LocationInput from './LocationInput';
-import { isMapsApiEnabled } from '../featureFlags';
-import {
-  getGeodataForPlace,
-  getGeodataForString,
-  getLatLng,
-} from '../services/GeoService';
-import parseDoc from '../util/parseDoc';
+import { getGeodataForPlace, getGeodataForString } from '../services/GeoService';
 import useQuery from '../util/useQuery';
+import { getInitialDocuments, searchDocuments, loadMoreDocuments } from '../services/loadData';
 
-export default function EntryList({ pageSize = 0 }) {
+export default function EntryList({ pageSize }) {
   const { t } = useTranslation();
   const history = useHistory();
   const windowLocation = useLocation();
 
   const [searching, setSearching] = useState(false);
   const [location, setLocation] = useState('');
+  const [placeId, setPlaceId] = useState();
   const [radius, setRadius] = useState(10); // In KM
   const [sliderVisible, setSliderVisible] = useState(false);
   const [entries, setEntries] = useState([]);
 
-  const collection = fb.store.collection('ask-for-help');
-  const geoCollection = new GeoFirestore(fb.store).collection('ask-for-help');
-
   const [lastEntry, setLastEntry] = useState(undefined);
-  const [scheduledSearch, setScheduledSearch] = useState(undefined);
 
   const { address: addressFromUrl } = useQuery();
 
-  const buildQuery = async (lastLoaded = undefined) => {
-    let query = collection.orderBy('d.timestamp', 'desc');
-
-    if (lastLoaded !== undefined) {
-      query = query.startAfter(lastLoaded);
-    }
-    if (pageSize > 0) {
-      query = query.limit(pageSize);
+  const appendDocuments = (documents, clear = false) => {
+    if (documents.length === 0) {
+      if (clear) {
+        setEntries([]);
+      }
+      return;
     }
 
-    return query;
-  };
-
-  const buildFilteredQuery = async (searchAttr, placeId = undefined) => {
-    // Fallback
-    if (!searchAttr) return buildQuery();
+    const oldestEntry = documents.filter((d) => !d.solved).reduce((oldest, doc) => (doc.timestamp < oldest.timestamp ? doc : oldest), documents[0]);
```

Comment:
```suggestion
    const oldestEntry = documents
    .filter((d) => !d.solved)
    .reduce((oldest, doc) => (doc.timestamp < oldest.timestamp ? doc : oldest), documents[0]);
```

 Makes it easier to read. Also, could we add a comment about why we filter the solved posts in the first place? To make sure we still  understand this in 6 months :)